includes/core/DataTransferObjects/API/class-response.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Response DTO.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Core\DataTransferObjects;
+
+use WCPay\Core\Contracts\API\Response\Base_Response;
+use WP_Http_Cookie;
+
+/**
+ * DTO for parsing response from WCPay server API.
+ */
+final class Response extends Data_Transfer_Object implements Base_Response {
```

Comment:
```php
... Response ... Base_Response
```

Is this something we plan to use for other external systems? If this is to be solely catered to the communication with the server, then shouldn't that we reflected in the class names?

Reply:
Maybe we could rename a class or put it inside an additional folder to display that we are actually referring to wcpay server. Our whole communication is done from the server and as it seems, the only communication will be from the server, so that's why I put naming like this. But putting classes into separate folder or change naming could also improve readability and extensibility if we decide to have more 'server' providers.